### ISSUE TITLE
[BUG] Error invoking CLI: `ImportError: cannot import name 'NotImplementedType' from 'types'

### ISSUE THREAD

**Issue by ReubenFrankel** [datnguye/dbterd#96]
**Describe the bug**
I get the following error invoking the `dbterd` CLI:

```
Traceback (most recent call last):
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/bin/dbterd", line 5, in <module>
    from dbterd.main import main
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbterd/main.py", line 1, in <module>
    from dbterd.cli import main as cli
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbterd/cli/main.py", line 6, in <module>
    from dbterd.adapters.base import Executor
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbterd/adapters/base.py", line 3, in <module>
    from types import NotImplementedType
ImportError: cannot import name 'NotImplementedType' from 'types' (/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/python3.9/types.py)
```

**To Reproduce**

```sh
python3.9 -m venv .venv
source .venv/bin.activate
pip install dbterd==1.12.0
dbterd --help
```

**Expected behavior**
To be able to use the CLI.

**Desktop**
- OS: Ubuntu 22.10
- Version: `1.12.0`

**Additional context**
I also tried `1.11.0`

```
Traceback (most recent call last):
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/bin/dbterd", line 5, in <module>
    from dbterd.main import main
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbterd/main.py", line 1, in <module>
    from dbterd.cli import main as cli
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbterd/cli/main.py", line 6, in <module>
    from dbterd.adapters.base import Executor
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbterd/adapters/base.py", line 9, in <module>
    from dbterd.adapters.dbt_cloud.administrative import DbtCloudArtifact
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbterd/adapters/dbt_cloud/administrative.py", line 4, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
```

After installing `requests` manually

```
Traceback (most recent call last):
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/bin/dbterd", line 5, in <module>
    from dbterd.main import main
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbterd/main.py", line 1, in <module>
    from dbterd.cli import main as cli
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbterd/cli/main.py", line 6, in <module>
    from dbterd.adapters.base import Executor
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbterd/adapters/base.py", line 9, in <module>
    from dbterd.adapters.dbt_cloud.administrative import DbtCloudArtifact
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbterd/adapters/dbt_cloud/administrative.py", line 6, in <module>
    from dbterd.helpers import file
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbterd/helpers/file.py", line 5, in <module>
    from dbt_artifacts_parser import parser
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbt_artifacts_parser/parser.py", line 20, in <module>
    from dbt_artifacts_parser.parsers.manifest.manifest_v1 import ManifestV1
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/dbt_artifacts_parser/parsers/manifest/manifest_v1.py", line 17, in <module>
    class ManifestMetadata(BaseParserModel):
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/pydantic/_internal/_model_construction.py", line 178, in __new__
    set_model_fields(cls, bases, config_wrapper, types_namespace)
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/pydantic/_internal/_model_construction.py", line 452, in set_model_fields
    fields, class_vars = collect_model_fields(cls, bases, config_wrapper, types_namespace, typevars_map=typevars_map)
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/pydantic/_internal/_fields.py", line 122, in collect_model_fields
    type_hints = get_cls_type_hints_lenient(cls, types_namespace)
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/pydantic/_internal/_typing_extra.py", line 212, in get_cls_type_hints_lenient
    hints[name] = eval_type_lenient(value, globalns, localns)
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/pydantic/_internal/_typing_extra.py", line 224, in eval_type_lenient
    return eval_type_backport(value, globalns, localns)
  File "/home/reuben/.cache/pypoetry/virtualenvs/dbterd-ext-IFoOWQ5l-py3.9/lib/python3.9/site-packages/pydantic/_internal/_typing_extra.py", line 240, in eval_type_backport
    return typing._eval_type(  # type: ignore
  File "/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/python3.9/typing.py", line 292, in _eval_type
    return t._evaluate(globalns, localns, recursive_guard)
  File "/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/python3.9/typing.py", line 554, in _evaluate
    eval(self.__forward_code__, globalns, localns),
  File "<string>", line 1, in <module>
TypeError: constr() got an unexpected keyword argument 'regex'
```

**Comment by datnguye** [comment]
Thanks @ReubenFrankel for finding it! 

I can reproduce it now with v1.12 -- will fix it soon 🏃 

For the v1.11, I think I have not fully tested standalone, it should be working with installing dbt-core -- so a workaround here:
```
pip install dbt-core
pip install dbterd=1.11.1
```
I will tackle this together and provide patches for all impacted version soon 🏃

**Comment by ReubenFrankel** [comment]
Is there a reason that `dbt-core` is not a defined dependency?

**Comment by datnguye** [comment]
Hey @ReubenFrankel yes, dbterd is originally designed to understand dbt artifact files and produce the ERD code. It doesn’t call any APIs from dbt package, it hence should not depend on dbt pip package.

**Comment by ReubenFrankel** [comment]
@datnguye Sure, but what about `requests` as in above example?

**Comment by datnguye** [comment]
@ReubenFrankel yeah I will add an extra deps with ‘requests’ for sure, and I need to fully verify all python versions to understand the case.

**Comment by datnguye** [comment]
It should be fixed in 1.12.2 now 🙌
